game/src/common/mod.rs

```diff
@@ -410,3 +411,60 @@ pub fn checkbox_per_mode(
     }
     Widget::custom_row(filters)
 }
+
+/// Many game states fit a pattern of managing a single panel, handling mouseover events, and other
+/// interactions on the map. Implementing this instead of `State` reduces some boilerplate.
+pub trait SimpleState {
+    /// Called when something on the panel has been clicked.
+    fn on_click(&mut self, ctx: &mut EventCtx, app: &mut App, x: &str, panel: &Panel)
+        -> Transition;
+    /// Called when something on the panel has changed.
+    fn panel_changed(&mut self, _: &mut EventCtx, _: &mut App, _: &Panel) -> Option<Transition> {
+        None
+    }
+    /// Called when the mouse has moved.
+    fn on_mouseover(&mut self, _: &mut EventCtx, _: &mut App) {}
+    /// Always called if the previous methods have not terminated.
```

Comment:
What does "terminated" mean in this context? 

It looks to me like we'll call `other_event` regardless of what happens in `on_mouseover`, so maybe referring to "the previous methods" should be "if `on_click or `panel_changed`"

Reply:
Good point! Updated the comment